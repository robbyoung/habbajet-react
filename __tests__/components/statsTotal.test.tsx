import React from 'react';
import renderer from 'react-test-renderer';
import StatsTotal from '../../app/components/statsTotal';

describe('StatsTotal Component', () => {
    it('can render a total label for the stats screen', () => {
        const component = renderer.create(
            <StatsTotal
                stats={[
                    {
                        total: 10,
                        color: '#ffffff',
                        percentage: 0.1,
                        tagName: 'testTag1',
                    },
                    {
                        total: 90,
                        color: '#000000',
                        percentage: 0.9,
                        tagName: 'testTag2',
                    },
                ]}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
