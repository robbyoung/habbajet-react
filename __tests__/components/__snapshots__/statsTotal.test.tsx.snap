// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsTotal Component can render a total label for the stats screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Abel",
          "fontSize": 20,
          "margin": 0,
          "padding": 0,
        },
        Object {
          "color": "#000000",
        },
      ]
    }
  >
    Total
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Abel",
          "marginTop": -5,
          "padding": 0,
        },
        Object {
          "fontSize": 50,
        },
      ]
    }
  >
    $100.00
  </Text>
</View>
`;
